x-pack/plugins/index_management/__jest__/client_integration/home/indices_tab.test.ts

```diff
@@ -12,6 +12,33 @@ import { setupEnvironment, nextTick } from '../helpers';
 import { IndicesTestBed, setup } from './indices_tab.helpers';
 import { createDataStreamPayload, createNonDataStreamIndex } from './data_streams_tab.helpers';
 
+// Since the editor component being used for editing index settings is not a React
+// component but an editor being instantiated on a div reference, we cannot mock
+// the component and replace it with something else. In this particular case we're
+// mocking the returned instance of the editor to always have the same values.
+jest.mock('../../../public/application/lib/ace.js', () => {
```

Comment:
I couldn't figure out what was the best way of testing this, the way the editor is instantiated is through [a div ref](https://github.com/elastic/kibana/blob/c72eb357650de4a90399420e015c9d905df3738a/x-pack/plugins/index_management/public/application/lib/ace.js#L47) which makes it hard to mock other than creating a ghost instance such as this one I implemented.

Reply:
Makes sense to me. Until we have a dedicated component we can mock this is the best we can do IMO.

What do you think about having the option to set the returned value from the ace editor from inside the test. It would make it clearer why in this case we return an invalid tier preference.

Something like this

```js
const mockGetAceEditorValue = jest.fn().mockReturnValue(`{}`);

jest.mock('../../../public/application/lib/ace.js', () => {
  const createAceEditor = () => {
    return {
      getValue: mockGetAceEditorValue,
      getSession: () => {
     ...

// And then in the specific test

test('shows error callout when request fails', async () => {
  const { actions, find, component, exists } = testBed;

  mockGetAceEditorValue.mockReturnValue(`{
    "index.routing.allocation.include._tier_preference": "non_existent_tier"
  }`);

...
```